src/modules/course/schema/enrolledCourse.schema.ts

```diff
@@ -0,0 +1,67 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import * as mongoose from 'mongoose';
+
+export type EnrolledCourseDocument = EnrolledCourse & Document;
+
+interface video {
+  num: number;
+  timestamp?: Date;
+}
+
+@Schema({
+  toJSON: { virtuals: true, getters: true },
+  toObject: { virtuals: true, getters: true },
+})
+export class EnrolledCourse {
+  @Prop({})
+  studentId: mongoose.Schema.Types.ObjectId;
+
+  @Prop({})
+  videos_watched: boolean[];
+
+  @Prop({})
+  assignmentsDone: boolean[];
+
+  @Prop({})
+  currentVideo: video[];
+
+  @Prop({})
+  doubts: string[];
+
+  @Prop({})
+  courseId: mongoose.Schema.Types.ObjectId;
+}
+
+export const EnrolledCourseSchema = SchemaFactory.createForClass(
+  EnrolledCourse,
+);
+
+// Pre hook to set default values
+EnrolledCourseSchema.pre('save', function (next) {
+  // eslint-disable-next-line @typescript-eslint/no-this-alias
+  const course = this;
+
+  // finds the length of the videos watched and gives it a default value
+  course['videos_watched'] = [true, true, false];
+  course['videos_watched'].fill(false);
```

Comment:
You are still filling the hardcoded value.
Get the number of videos from course and fill the array wit all `false`.